### PR TITLE
Reset analytics user identity when user gets logged out

### PR DESCRIPTION
**Pull request by vinaysshenoy**
Pivotal Story Link: https://www.pivotaltracker.com/story/show/166151512

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -31,6 +31,10 @@
     <meta-data
       android:name="google_analytics_adid_collection_enabled"
       android:value="false" />
+    <meta-data
```
Comment (vinaysshenoy) [inline app/src/main/AndroidManifest.xml]: The build started complaining unless I added this. I had to poke around in the merged manifest to figure out what went wrong. I tested a build to make sure analytics are still getting pushed, and it seems like it's all working fine.

**Comment by vinaysshenoy** [comment]
> Please fix conflicting files @vinaysshenoy

Done @rakshakhegde

File: app/src/test/java/org/simple/clinic/analytics/AnalyticsTest.kt [hunk 2]
```diff
@@ -147,14 +152,32 @@ class AnalyticsTest {
...
+  }
+
+  @Test
+  fun `when clearing the user id, the user ID must be cleared from the reporters`() {
+    val reporter = MockAnalyticsReporter()
+    Analytics.addReporter(reporter)
+
+    val uuid = UUID.randomUUID()
+    Analytics.setUserId(uuid)
+    assertThat(reporter.setUserId).isEqualTo(uuid.toString())
```
Comment (rakshakhegde) [inline app/src/test/java/org/simple/clinic/analytics/AnalyticsTest.kt]: `setUserId` is a property. Can this be renamed to just `userId`?

File: app/src/main/java/org/simple/clinic/analytics/Analytics.kt [hunk 3]
```diff
@@ -33,7 +33,14 @@ object Analytics {
         val uuidString = uuid.toString()
 
         setUserIdentity(uuidString)
-        setProperty("userId", uuidString)
+      }
```
Comment (rakshakhegde) [inline app/src/main/java/org/simple/clinic/analytics/Analytics.kt]: I see this is not part of your commits, but Android Studio IDE warns us that:
image

Can you please fix this on lines 19 and 27?

File: app/src/test/java/org/simple/clinic/analytics/AnalyticsTest.kt [hunk 4]
```diff
@@ -167,8 +190,13 @@ class AnalyticsTest {
     val userId = UUID.randomUUID()
```
Comment (rakshakhegde) [inline app/src/test/java/org/simple/clinic/analytics/AnalyticsTest.kt]: Shouldn't the name of the test say "...user id must ~by~ **be** set on..."?

File: app/src/test/java/org/simple/clinic/analytics/AnalyticsTest.kt (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/analytics/AnalyticsTest.kt]: My intention was to capture the meaning `the user ID that was set on this reporter`. I guess `userId` will work just as well and won't clash with the convention of the Java setter method.

File: app/src/test/java/org/simple/clinic/analytics/MockAnalyticsReporter.kt [hunk 5]
```diff
@@ -2,38 +2,32 @@ package org.simple.clinic.analytics
...
 
-  fun clearReceivedEvents() {
-    receivedEvents.clear()
-  }
-
-  fun clearSetProperties() {
-    setProperties.clear()
+  fun clear() {
+    clearReceivedEvents()
+    clearSetUserIds()
```
Comment (rakshakhegde) [inline app/src/test/java/org/simple/clinic/analytics/MockAnalyticsReporter.kt]: `clearUserIds()`?

File: app/src/main/java/org/simple/clinic/analytics/Analytics.kt (hunk 3 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/analytics/Analytics.kt]: Done.

File: app/src/test/java/org/simple/clinic/analytics/AnalyticsTest.kt (hunk 4 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/analytics/AnalyticsTest.kt]: Whoops, will fix/

File: app/src/test/java/org/simple/clinic/analytics/MockAnalyticsReporter.kt (hunk 5 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/analytics/MockAnalyticsReporter.kt]: Done.

**Pull request merged**